sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionCatalog.scala

```diff
@@ -83,29 +103,30 @@ private[sql] class HiveSessionCatalog(
   // essentially a cache for metastore tables. However, it relies on a lot of session-specific
   // things so it would be a lot of work to split its functionality between HiveSessionCatalog
   // and HiveCatalog. We should still do it at some point...
-  private val metastoreCatalog = new HiveMetastoreCatalog(sparkSession)
-
-  val ParquetConversions: Rule[LogicalPlan] = metastoreCatalog.ParquetConversions
-  val OrcConversions: Rule[LogicalPlan] = metastoreCatalog.OrcConversions
-  val CreateTables: Rule[LogicalPlan] = metastoreCatalog.CreateTables
+  private val metadataCache = new LogicalRelationCache(sparkSession)
 
   override def refreshTable(name: TableIdentifier): Unit = {
     super.refreshTable(name)
-    metastoreCatalog.refreshTable(name)
+    metadataCache.refreshTable(name)
   }
 
-  def invalidateCache(): Unit = {
-    metastoreCatalog.cachedDataSourceTables.invalidateAll()
+  override def invalidateCache(): Unit = {
+    metadataCache.invalidateAll()
   }
 
-  def hiveDefaultTableFilePath(name: TableIdentifier): String = {
-    metastoreCatalog.hiveDefaultTableFilePath(name)
+  override def cacheDataSourceTable(name: TableIdentifier, plan: LogicalPlan): Unit = {
```

Comment:
instead of having these 3 new APIs, can we just expose the `logicalRelationCache`?